### PR TITLE
Setup PR Preview workflow

### PR DESCRIPTION
**Pull request by lsetiawan**
## Overview

This PR uses https://surge.sh/ for PR preview. Currently this will not work as I don't have enough permission to add `SURGE_TOKEN` into secrets. Someone needs to create the token and set it. Please see instructions there: https://surge.sh/help/integrating-with-travis-ci

### CODE REVIEW

**Comment by scottyhq** [comment]
Nice! This would close #68. I also don't have access to repo secrets I think @dcherian or @andersy005 would need to add `SURGE_TOKEN`

**Comment by dcherian** [comment]
@andersy005 can you take a look please?

File: .github/workflows/preview.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  pull-requests: write # allow surge-preview to create/update PR comments
+
+jobs:
+  preview:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - uses: ./.github/actions/setupconda
```
Comment (andersy005) [inline .github/workflows/preview.yaml:15]:
```suggestion
       - name: Setup JupyterBook Cache
        uses: actions/cache@v3
        with:
          path: _build
          # NOTE: change key to "jupyterbook-DATE" to force rebuilding cache
          key: jupyterbook-20230602

      - name: Install Conda environment with Micromamba
        uses: mamba-org/setup-micromamba@v1
        with:
          environment-file: conda/conda-lock.yml
          environment-name: xarray-tutorial
          cache-environment: true
```

File: .github/workflows/preview.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      - name: Dump Build Logs
+        if: always()
+        run: |
+          if (test -a _build/html/reports/*log); then cat _build/html/reports/*log ; fi
+
+      - name: Publish to Surge
+        id: preview_step
+        uses: afc163/surge-preview@v1
+        with:
+          surge_token: ${{ secrets.SURGE_TOKEN }}
```
Comment (andersy005) [inline .github/workflows/preview.yaml:32]: @lsetiawan, i just added this to the repo secrets.

Comment (lsetiawan) [reply .github/workflows/preview.yaml:32]: Thank you so much! And thanks for the comments, I'll take a look and make the changes 🙏🏽

**Comment by lsetiawan** [comment]
This is ready to go now. Seems like the PR comments only works for non-forked PRs. This is due to the fact that Github has restricted PR permission to read-only for forked PRs. See https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

**Comment by lsetiawan** [comment]
To access the URL to the preview, maintainers can go to the workflow summary. For example: https://github.com/xarray-contrib/xarray-tutorial/actions/runs/5260269206

**Comment by dcherian** [comment]
Quick Q: What's the reason for preferring this workflow over the RTD workflow?

**Comment by scottyhq** [comment]
> Quick Q: What's the reason for preferring this workflow over the RTD workflow? 

I've helped out with a curriculum book using RTD here https://github.com/UW-GDA/jupyterbook. The PR previews and versioning is definitely nice, but the main drawback in my mind is navigating over to a separate web interface for exploring logs when things go wrong. Also building via github actions gives more flexibility (and perhaps more resources?) but that is not such a big deal if the notebooks remain small

**Comment by dcherian** [comment]
My concern is the extra (unfamiliar) maintenance overhead, but we can evaluate later.

**Pull request merged**